pymodbus/server/simulator/http_server.py

```diff
@@ -301,14 +302,18 @@ async def handle_html(self, request):
     async def handle_json(self, request):
         """Handle api registers."""
         page_type = request.path.split("/")[-1]
-        params = await request.post()
-        json_dict = self.generator_html[page_type][0].copy()
-        result = self.generator_json[page_type][1](params, json_dict)
-        return web.Response(text=f"json build: {page_type} - {params} - {result}")
+        params = await request.json()
```

Comment:
this looks very complicated, is sounds as if removing _json could simplify this a lot.

Reply:
From your immediate view, I get it, and kind of agree with it. From the point of anyone automating anything based on this, not so much. But please let me know if you plan to scrap it.

Also, I think many of the things could be streamlined with changing the structure of how these are handled. IMHO the only complexity comes from the current copy-paste model I adopted, as to not break anything else, and possibly to have this discussion. Changing the logic to have the data and encoding separate, preferrably with automatic serialization with json could help alot. But I felt it would have warranted too much changes to the existing and been too intrusive.

Reply:
It is normal to have multiple endpoints for different types of service, so we clearly want 1 endpoint for html and another for RestAPI.

We do NOT want to use the same endpoint for different services and distinguish in the naming !